### PR TITLE
Improve README.md docker instructions

### PR DESCRIPTION
**Pull request by sarenameas**
- Update ngrok instructions to be on separate lines.
- Fix updating the mqtt_demo_plaintext server address after setting up the plaintext server.'
- Use relative paths for links to files in this repo.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -303,14 +303,14 @@ The following instructions have been tested on an Ubuntu 18.04 environment with
...
 
-    `BROKER_ENDPOINT` defined in `demos/mqtt/mqtt_demo_basic_tls/demo_config.h` can now be set to `localhost`.
-
 1. If a Mosquitto broker with TLS communication needs to be run, ignore this step and proceed to the next step. A Mosquitto broker with plain text communication can be run by executing the command below.
 
     ```
     docker run -it -p 1883:1883 --name mosquitto-plain-text eclipse-mosquitto:latest
     ```
 
+1. Set `BROKER_ENDPOINT` defined in `demos/mqtt/mqtt_demo_plaintext/demo_config.h` to `localhost`.
```
Comment (sarenameas) [inline README.md:312]:
```suggestion
1. Set `BROKER_ENDPOINT` defined in [demos/mqtt/mqtt_demo_plaintext/demo_config.h](demos/mqtt/mqtt_demo_plaintext/demo_config.h) to `localhost`.
```

File: README.md [hunk 2]
```diff
@@ -349,7 +349,9 @@ The following instructions have been tested on an Ubuntu 18.04 environment with
     docker run -it -p 8883:8883 -v $(pwd):/mosquitto/config/ --name mosquitto-basic-tls eclipse-mosquitto:latest
     ```
 
-1. Set `ROOT_CA_CERT_PATH` to the absolute path of the CA certificate created in step 3. for the local Mosquitto server.
+1. Update `demos/mqtt/mqtt_demo_basic_tls/demo_config.h` to the following:
```
Comment (sarenameas) [inline README.md:352]:
```suggestion
1. Update [demos/mqtt/mqtt_demo_basic_tls/demo_config.h](demos/mqtt/mqtt_demo_basic_tls/demo_config.h) to the following:  
```

File: README.md [hunk 3]
```diff
@@ -361,19 +363,22 @@ docker run -p 80:80 kennethreitz/httpbin
...
+
+Set `SERVER_HOST` in `demos/http/http_demo_basic_tls/demo_config.h` to the https link provided by ngrok, without `https://` preceeding it.
+
+You must also download the Root CA certificate provided by the ngrok https link and set `ROOT_CA_CERT_PATH` in `demo_config.h` to the file path of the downloaded certificate.
 
 ## Generating Documentation
 
-The Doxygen references were created using Doxygen version 1.8.20. To generate the Doxygen pages, use the provided Python script, [tools/doxygen/generate_docs.py](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/tools/doxygen/generate_docs.py). Please ensure that each of the library submodules under libraries/standard/ and libraries/aws are cloned before using this script.
+The Doxygen references were created using Doxygen version 1.8.20. To generate the Doxygen pages, use the provided Python script in [tools/doxygen/generate_docs.py](tools/doxygen/generate_docs.py). Please ensure that each of the library submodules under libraries/standard/ and libraries/aws are cloned before using this script.
```
Comment (sarenameas) [inline README.md:381]:
```suggestion
The Doxygen references were created using Doxygen version 1.8.20. To generate the Doxygen pages, use the provided Python script at [tools/doxygen/generate_docs.py](tools/doxygen/generate_docs.py). Please ensure that each of the library submodules under libraries/standard/ and libraries/aws are cloned before using this script.
```

File: README.md [hunk 4]
```diff
@@ -303,14 +303,14 @@ The following instructions have been tested on an Ubuntu 18.04 environment with
     docker pull eclipse-mosquitto:latest
     ```
 
-    `BROKER_ENDPOINT` defined in `demos/mqtt/mqtt_demo_basic_tls/demo_config.h` can now be set to `localhost`.
```
Comment (sarenameas) [inline README.md:306]: This was incorrect

**Review by jainamt — APPROVED** [review]

**Review by jainamt — APPROVED** [review]

**Review by jainamt — DISMISSED** [review]

**Review by archigup — APPROVED** [review]

**Review by leegeth — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**